### PR TITLE
Fix DJ001 Disallow null CharField on BaseZarrArchive.checksum

### PR DESCRIPTION
**Pull request by brianhelba**
See: https://docs.astral.sh/ruff/rules/django-nullable-model-string-field/

### CODE REVIEW

File: dandiapi/zarr/models.py [hunk 1]
```diff
@@ -39,22 +39,22 @@ class Meta:
...
-                    checksum__isnull=True,
+                    checksum='',
                     status__in=[
                         ZarrArchiveStatus.PENDING,
                         ZarrArchiveStatus.UPLOADED,
                         ZarrArchiveStatus.INGESTING,
                     ],
                 )
-                | models.Q(checksum__isnull=False, status=ZarrArchiveStatus.COMPLETE),
+                | (~models.Q(checksum='') & models.Q(status=ZarrArchiveStatus.COMPLETE)),
```
Comment (brianhelba) [inline dandiapi/zarr/models.py:49]: I'm not sure if there's a less awkward way to combine these conditions?

File: dandiapi/zarr/tests/test_ingest_zarr_archive.py [hunk 2]
```diff
@@ -88,7 +88,7 @@ def test_ingest_zarr_archive_assets(zarr_archive_factory, zarr_file_factory, dra
     # Assert asset size, metadata
     assert asset.size == 0
     assert asset.full_metadata['contentSize'] == 0
-    assert asset.full_metadata['digest']['dandi:dandi-zarr-checksum'] is None
+    assert asset.full_metadata['digest']['dandi:dandi-zarr-checksum'] == ''
```
Comment (brianhelba) [inline dandiapi/zarr/tests/test_ingest_zarr_archive.py:91]: Note, this change does leak out via the HTTP API, as shown. I'm not sure if that's a compatibility problem with anything.

Comment (AlmightyYakob) [reply dandiapi/zarr/tests/test_ingest_zarr_archive.py:91]: This will most likely require a corresponding change in the dandi CLI

**Comment by brianhelba** [comment]
Closing. Will just `#noqa` this issue.

**Pull request closed without merging**